packages/component/src/Activity/Avatar.js

```diff
@@ -1,24 +1,66 @@
 import classNames from 'classnames';
+import memoize from 'memoize-one';
 import React from 'react';
 
 import connectToWebChat from '../connectToWebChat';
 
-export default connectToWebChat(
-  ({ styleSet }) => ({ styleSet })
-)(
+const connectAvatar = (...selectors) => connectToWebChat(
   ({
-    children,
-    className,
-    fromUser,
-    styleSet
-  }) =>
-    <div
-      className={ classNames(
-        styleSet.avatar + '',
-        { 'from-user': fromUser },
-        (className || '') + ''
-      ) }
-    >
-      { children }
-    </div>
-)
+    styleSet: {
+      options: {
+        botAvatarImage,
+        botAvatarInitials,
+        userAvatarImage,
+        userAvatarInitials
+      }
+    }
+  }, { fromUser }) => ({
+    avatarImage: fromUser ? userAvatarImage : botAvatarImage,
+    avatarInitials: fromUser ? userAvatarInitials : botAvatarInitials
+  }),
+  ...selectors
+);
+
+class Avatar extends React.Component {
+  constructor() {
+    super();
+
+    this.createBackgroundImageStyle = memoize(backgroundImage => ({ backgroundImage: `url(${ encodeURI(backgroundImage) })` }));
+  }
+
+  render() {
+    const {
+      props: {
+        avatarImage,
+        avatarInitials,
+        className,
+        fromUser,
+        styleSet
+      }
+    } = this;
+
+    return (
+      !!(avatarImage || avatarInitials) &&
+        <div
+          className={ classNames(
+            styleSet.avatar + '',
+            { 'from-user': fromUser },
+            (className || '') + ''
+          ) }
+        >
+          { avatarInitials }
+          <div
```

Comment:
We use avatar initials as a way to provide alt text, otherwise, the user agent will say it twice.

`object-fit` in IE11 is a good call. We have a `<CroppedImage>` component to do the object-fit work.

Reply:
Moved to our `<CroppedImage>`, it use real `<img>` and works like `object-fit: cover`.

The alt text is removed because otherwise it will dupe with initials, i.e. screen reader will say "WC WC". I have another tracking bug #1499 for the screen reader to read "Initials: WC" other than "WC".